threatgrid/ctia
# Allow `added` and `deleted` fields in update change event (Take 2)

Close https://github.com/threatgrid/iroh/issues/3551

Update limits are not handled and are deferred to a future fix of https://github.com/threatgrid/iroh/issues/3680.

- [x] test this actually solves https://github.com/threatgrid/iroh/issues/3551
- [x] move test to own ns
- [x] fix query in test





<a name="qa">[§](#qa)</a> QA
============================

This is a general fix for a specific problem, in particular for missing `:assignees` in update events. See attached issue https://github.com/threatgrid/iroh/issues/3551

There are three interesting cases to verify this fixes the attached issue.

1. Add an `assignees` field to an `Incident` which doesn't already have one. Verify the resulting event contains an `added` entry similar to this in its `:fields`:

```clojure
      {
        "field": "assignees",
        "action": "added",
        "change": {
          "after": <your-assignees>
        }
      }
```

2. Modify an existing `assignees` field in an `Incident. Verify the resulting event contains a `modified` entry similar to this in its `:fields`:

```clojure
      {
        "field": "assignees",
        "action": "modified",
        "change": {
          "before": <your-assignees-before-change>
          "after": <your-assignees-after-change>
        }
      }
```

3. Delete an existing `assignees` field in an `Incident. Verify the resulting event contains a `deleted` entry similar to this in its `:fields`:

```clojure
      {
        "field": "assignees",
        "action": "deleted",
        "change": {
          "before": <your-assignees-before-deletion>
        }
      }
```





<a name="release-notes">[§](#release-notes)</a> Release Notes
=============================================================



```
intern: Allow `added` and `deleted` fields in update change event
```

<a name="squashed-commits">[§](#squashed-commits)</a> Squashed Commits
======================================================================
